hive/agents/td3.py

```diff
@@ -127,6 +127,8 @@ def __init__(
         self._action_max = self._action_space.high
         self._action_scaling = 0.5 * (self._action_max - self._action_min)
         self._scale_actions = np.isfinite(self._action_scaling).all()
+        self._action_min_tensor = torch.as_tensor(self._action_min, device=self._device)
```

Comment:
@dapatil211  I think you can just keep self._action_min_tensor and maybe change all numpy operations that use the self._action_min to torch operations, and then change the action tensor to numpy using .numpy() [as opposed to first converting tensor to numpy and then using numpy operations]

I believe this will improve readability

Reply:
Actually I just realized why I didn't do this. When I put the scale the action before putting it into the replay buffer, it's a numpy array. So I'd have to convert it to a torch tensor (on device) and then back to a numpy array.

Reply:
I get the issue, so it seems we need to have a different variable.

Another point of improvement is that in the framework actions which are generated from the agent are passed to the runner and then are received again in the dict. Maybe we should avoid this redundancy?

Reply:
I get the issue, so it seems we need to have a different variable.

Another point of improvement is that in the framework actions which are generated from the agent are passed to the runner and then are received again in the dict. Maybe we should avoid this redundancy?

Reply:
I think for now it's ok to have that. Especially because to implement the parallel env stuff, i am planning on making the agents stateless.